pkg/provisioner/ironic/ironic.go

```diff
@@ -1176,9 +1176,14 @@ func (p *ironicProvisioner) Delete() (result provisioner.Result, err error) {
 func (p *ironicProvisioner) changePower(ironicNode *nodes.Node, target nodes.TargetPowerState) (result provisioner.Result, err error) {
 	p.log.Info("changing power state")
 
-	// If we're here, we're going to change the state and we should
-	// wait to try that again.
-	result.RequeueAfter = powerRequeueDelay
+	switch nodes.ProvisionState(ironicNode.ProvisionState) {
+	case nodes.Inspecting, nodes.InspectWait, nodes.InspectFail, nodes.Cleaning, nodes.CleanWait:
```

Comment:
There is an easier way to find it out: inspect the `node.target_provision_state` field. If it's not `null`, you probably cannot (and definitely should not) change its power state.

Reply:
That's much simpler, yes.